fe/fe-core/src/main/java/com/starrocks/connector/partitiontraits/IcebergPartitionTraits.java

```diff
@@ -64,7 +65,7 @@ public List<PartitionInfo> getPartitions(List<String> partitionNames) {
     @Override
     public Optional<Long> maxPartitionRefreshTs() {
         IcebergTable icebergTable = (IcebergTable) table;
-        return icebergTable.getSnapshot().map(Snapshot::timestampMillis);
+        return Optional.ofNullable(icebergTable.getNativeTable().currentSnapshot()).map(Snapshot::timestampMillis);
```

Comment:
This may cause snapshot is inconsistent for mv

Reply:
I found that mv always use the latest snapshot in the query.  or mv should record the version range for rewriting.